.github/workflows/deploy-apidocs.yml

```diff
@@ -37,37 +37,48 @@ jobs:
       - name: Setup PHP
         uses: shivammathur/setup-php@v2
         with:
+          # phpDocumentor v3.0 does not have support yet for PHP 8.0
           php-version: '7.4'
           extensions: intl
 
       - name: Install GraphViz for phpDocumentor
         run: |
-          sudo apt install -yq graphviz
+          sudo apt-get install -yq graphviz
 
-      - name: Download phpDocumentor
+      - name: Download phpDocumentor v3.0
         run: |
           cd ./source
           curl \
-            -L https://github.com/phpDocumentor/phpDocumentor/releases/download/v2.9.1/phpDocumentor.phar \
+            -L https://github.com/phpDocumentor/phpDocumentor/releases/download/v3.0.0/phpDocumentor.phar \
             -o admin/phpDocumentor.phar
 
-      - name: Prepare
+      - name: Prepare API repo
+        working-directory: api
         run: |
-          cd ./api
-          rm -rf ./api/docs*
-          mkdir -p ./api/docs
+          rm -rf api/docs*
+          mkdir -p api/docs
           git reset --hard master
 
-      - name: Build
+      - name: Build API in source repo
+        working-directory: source
         run: |
-          cd ./source
-          chmod +x admin/phpDocumentor.phar
-          admin/phpDocumentor.phar
+          php admin/phpDocumentor.phar run --ansi --verbose
           cp -R ${GITHUB_WORKSPACE}/source/api/build/* ${GITHUB_WORKSPACE}/api/docs
 
-      - name: Deploy
+      - name: Deploy to API repo
+        working-directory: api
         run: |
-          cd ./api
           git add .
-          git commit -m "API sync for ${GITHUB_SHA}"
-          git push
+          git commit -S -m "Updated API for commit ${{ github.event.after }}"
```

Comment:
Are you sure this is the correct reference?

https://github.community/t/accessing-commit-message-in-pull-request-event/17158/8

Reply:
I have checked this but if that gotcha exists then it's better to use the sha directly. Reverted.